src/components/Button.js

```diff
@@ -1,12 +1,152 @@
+  color: "#2196f3",
+  background: "#b3e5fc"
+};
+
+const btnPrimaryGrayShadow ={
+  name :"btnPrimaryGrayShadow",
+  color: "#2196f3",
+  background: "#ffffff",
+  boxShadow: "0px 1px 10px rgba(0, 0, 0, 0.2), 0px 4px 5px rgba(0, 0, 0, 0.12), 0px 2px 4px rgba(0, 0, 0, 0.14)"
+};
+const btnPrimaryShadow = {
+  name:"btnPrimaryShadow",
+  color: "#2196f3",
+  background: "#ffffff",
+  boxShadow: "0px 2px 4px #03A9F4"
+
+};
+
+
+/* secondary buttons styling */
+const btnSecondary = {
+  name : "btnSecondary",
+  color: "#ffffff",
+  background: "#4CAF50"
+};
+const btnSecondaryOutline = {
+  name:"btnSecondaryOutline",
+   color: "#4CAF50",
+  background: "#ffffff",
+  border: "2px solid #4CAF50"
+};
+const btnSecondaryLight ={
+  name:"btnSecondaryLight",
+  color: "#388E3C",
+  background: "#81C784",
+  opacity: "0.8"
+};
+const btnSecondaryGrayShadow = {
+  name:"btnSecondaryGrayShadow",
+   color: "#388E3C",
+  background: "#ffffff",
+  boxShadow: "0px 1px 10px rgba(0, 0, 0, 0.2), 0px 4px 5px rgba(0, 0, 0, 0.12),0px 2px 4px rgba(0, 0, 0, 0.14)"
+};
+const btnSecondaryShadow ={
+  name : "btnSecondaryShadow",
+  color: "#388E3C",
+  background: "#ffffff",
+  boxShadow: "0px 2px 4px #4CAF50"
+};
+
+
+/* error buttons styling */
+const btnError ={
+  name:"btnError",
+  color: "#ffffff",
+  background: "#f44336"
+};
+const btnErrorOutline ={
+  name :"btnErrorOutline",
+  color: "#f44336",
+  background: "#ffffff",
+  border: "2px solid #f44336"
+};
+const btnErrorLight={
+  name:"btnErrorLight",
+  color: "#f44336",
+  background: "#FFCDD2",
+  opacity: "0.8"
+};
+const btnErrorGrayShadow ={
+  name:"btnErrorGrayShadow",
+  color: "#f44336",
+  background: "#ffffff",
+  boxShadow: "0px 1px 10px rgba(0, 0, 0, 0.2), 0px 4px 5px rgba(0, 0, 0, 0.12),0px 2px 4px rgba(0, 0, 0, 0.14)"
+};
+const btnErrorShadow = {
+  name :"btnErrorShadow",
+  color: "#f44336",
+  background: "#ffffff",
+  boxShadow: "0px 2px 4px #F44336"  
+};
+//Add all buttons different  objects here 
+const buttonTypes = [
+  btnPrimary,
+  btnPrimaryOutlined,
+  btnPrimaryLight,
+  btnPrimaryGrayShadow,
+  btnPrimaryShadow,
+  btnSecondary,
+  btnSecondaryOutline,
+  btnSecondaryLight,
+  btnSecondaryGrayShadow,
+  btnSecondaryShadow,
+  btnError,
+  btnErrorOutline,
+  btnErrorLight,
+  btnErrorGrayShadow,
+  btnErrorShadow
+] 
+
+//Return the wanted button styles according the type props that send to the component
+const getSelectedButtonStyle =(type)=>buttonTypes.find(btn=>btn.name === type);
+
+//The button styled component 
+const Button = styled.button`
+   font-style: normal;
+  font-weight: 600;
+  font-size: 14px;
+  font-family: 'Open Sans', sans-serif;
+  line-height: 14px;
+   color: ${props=>props.type.color};
+  display: flex;
+  align-items: center;
+  text-align: center;
+  letter-spacing: 0.75px;
+  text-transform: capitalize;
+  padding: 8px 30px;
+  border-radius: 5px;
+  box-shadow : ${props=>props.type.background?props.type.boxShadow : "none"};
```

Comment:
I'll have to do some research, but instead of always changing a CSS property based on the type, I think we could spread the CSS for the different types in the Button component. Maybe look into the CSS helper from styled components. So we would have a base Button component and add specific styles depending on the variant.

Reply:
Thinking about this a bit more, it's probably even better if we just export different button components. Depends on what the team thinks also. 

So we could have:
```
BaseButton = styled.Button`
some base styles
`
export PrimaryButton = styled(BaseButton)`
${BaseButton}
...some extra styles
`

Reply:
Will have to discuss/think about this, but we can go with variant for now

Reply:
One thing I do see is that we'll have to many variants. I think we should just have variants for the actual buttons, i.e. primary, secondary, error, etc. And the different status of each variant can be done through props. Like shadow, outline, etc.

Reply:
The repeated variant is on the stories not in the actual code of the component  itself.